### PR TITLE
wasm: base64, json (un)marshal, and object.filter builtins.

### PR DESCRIPTION
**Pull request by koponen-styra**
Signed-off-by: Teemu Koponen <koponen@styra.com>

### CODE REVIEW

File: wasm/src/object.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            k = key;
+            break;
+        case OPA_ARRAY:
+            k = opa_value_get(keys, key);
+        }
+
+        opa_object_elem_t *elem = opa_object_get(opa_cast_object(obj), k);
+        if (elem != NULL)
+        {
+            opa_object_insert(r, k, elem->v);
```
Comment (tsandall) [inline wasm/src/object.c:35]: If we ever start emitting free() calls in the policy wasm we'll need to implement reference counting or similar to handle cases like this.

**Review by tsandall — DISMISSED** [review]
LGTM

**Comment by tsandall** [comment]
Looks like the generated wasm is not correct. Perhaps because of the builder changes in the regex PR?

File: wasm/src/object.c (hunk 1 above)
Comment (koponen-styra) [reply wasm/src/object.c:35]: Right, that seems like the way to go.

**Comment by koponen-styra** [comment]
> Looks like the generated wasm is not correct. Perhaps because of the builder changes in the regex PR?

Fixed.

**Review by tsandall — APPROVED** [review]

**Pull request merged**